src/modules/app/containers/RouterFragment.js

```diff
@@ -13,54 +13,80 @@ import PdfFetcherPage from 'routes/pdf-fetcher/containers/PdfFetcherPage'
 import MainDisclaimerPage from 'routes/main-disclaimer/components/MainDisclaimerPage'
 import LandingPage from 'routes/landing/containers/LandingPage'
 import CategoriesPage from 'routes/categories/containers/CategoriesPage'
+import I18nRedirect from 'modules/app/containers/I18nRedirect'
+import PropTypes from 'prop-types'
+import RouteConfig from '../../redux-little-router-config/RouteConfig'
 
 /**
  * todo: Test and document in WEBAPP-90
  * todo: Layouts should be set in each route
  */
 class RouterFragment extends React.Component {
+  static propTypes = {
+    routerConfig: PropTypes.instanceOf(RouteConfig).isRequired
+  }
+
+  static isLanguageCode (language) {
+    return language && language.length === 2
+  }
+
   render () {
+    const {routerConfig} = this.props
+    const matchRoute = (id) => routerConfig.matchRoute(id)
+
     /*
      * For routes inside a <React.Fragment /> the priority decreases with each element
      * So /disclaimer has higher priority than /:language -> '/disclaimer' resolves to /disclaimer
      */
     return <Fragment forRoute='/'>
       {/* Routes */}
       <React.Fragment>
+        {/* No language was provided to redirect to a specific language (e.g. the browsers language) */}
+        <Fragment forRoute='/'>
+          <I18nRedirect />
+        </Fragment>
 
         {/* Matches two or more arguments like /augsburg/de */}
         <Fragment forRoute='/:location/:language(/*)'>
-          <React.Fragment>
+          <LocationLayout matchRoute={matchRoute}>
             {/* Matches /augsburg/de/search -> Search */}
             <Fragment forRoute='/search'>
-              <LocationLayout><SearchPage /></LocationLayout>
+              <SearchPage />
             </Fragment>
             {/* Matches /augsburg/de/disclaimer -> Disclaimer */}
             <Fragment forRoute='/disclaimer'>
-              <LocationLayout><DisclaimerPage /></LocationLayout>
+              <DisclaimerPage />
             </Fragment>
             {/* Matches /augsburg/de/events* -> Events */}
             <Fragment forRoute='/events(/:id)'>
-              <LocationLayout><EventsPage /></LocationLayout>
+              <EventsPage />
             </Fragment>
             {/* Matches /augsburg/de/fetch-pdf/* -> Redirect */}
             <Fragment forRoute='/fetch-pdf'>
-              <Layout><PdfFetcherPage /></Layout>
+              <PdfFetcherPage />
```

Comment:
Uses now LocationLayout

Reply:
It shouldn't but, but it would make things a lot more complicated. I think we don't really need to care here.